yellow-paper/docs/decentralization/actors.md

```diff
@@ -0,0 +1,152 @@
+- Aztec nodes that are not running dependency-minimized can rely on third parties for body retrieval and use Ethereum as a "trust-anchor". Here meaning that the Aztec node could retrieve the full state from IPFS and then validate it against the headers it receives from the Ethereum node - allowing a quick synchronization process. Since the index is not directly part of state, its validity cannot be validated as simple as the state. The index will have to be built from the blocks (validating individual decrypted messages can be done against the state).
+
+An Aztec light-node is a bit different, since it does not store the actual state *nor* index, only the headers. This means that it will follow the same model of Ethereum light-nodes, where it can retrieve data by asking third parties for it and then validating that it is in the state (anchored by the headers stored). 
+
+:::info Aztec-specific Ultra Light nodes
+For users following the validating bridge as the canonical rollup, the bridge IS the light node, so they can get away with not storing anything, and just download the header data from the bridge. Since the [archive](./../state/archive.md) is stored in the bridge, an Aztec ultra-light node can validate membership proofs against it without needing the exact header (given sufficient membership proofs are provided). This essentially enable instant sync for ultra-light nodes -> if you either run an Ethereum node or trust a third party.
+:::
+
+### Responding to Queries
+
+In the following, we will briefly outline how the different type of Aztec node will respond to different kinds of queries. Namely we will consider the following queries:
+- Requesting current public state and membership proofs
+- Requesting historical public state and membership proofs
+- Requesting current membership proof of note hash, nullifier, l1 to l2 message or contract 
+- Requesting historical membership proof of note hash, nullifier, l1 to l2 message or contract 
+- Requesting encrypted notes for a given tag
+
+#### Light-Node
+
+As mentioned, we will be retrieving data from a third party, and then validating it against the headers at our disposal. Note that if we have the latest archive hash, a response can include an inclusion in a header and in the archive, thereby allowing us to validate the response against the archive as well.
+
+If we don't care about other people seeing what data we are retrieving, we can simply ask the third parties directly. However, when dealing with private information (e.g. notes), you need to be careful about how you retrieve it - you need to use some form of private information retrieval (PIR) service. 
```

Comment:
Here they wouldn't be retrieving the notes directly but rather the merkle path required to produce an inclusion proof, right?

Reply:
Extended the document a bit here as well.